File: dotnet/test/webdriverbackedselenium/TestLocators.cs
using NUnit.Framework;

namespace Selenium.Tests
{
    [TestFixture]
    public class TestLocators : SeleniumTestCaseBase
    {
        [Test]
        public void ShouldBeAbleToUseLocators()
        {
            selenium.Open("../tests/html/test_locators.html");
            // Id location
            Assert.AreEqual(selenium.GetText("id=id1"), "this is the first element");
            Assert.IsFalse(selenium.IsElementPresent("id=name1"));
            Assert.IsFalse(selenium.IsElementPresent("id=id4"));
            Assert.AreEqual(selenium.GetAttribute("id=id1@class"), "a1");
            // name location
            Assert.AreEqual(selenium.GetText("name=name1"), "this is the second element");
            Assert.IsFalse(selenium.IsElementPresent("name=id1"));
            Assert.IsFalse(selenium.IsElementPresent("name=notAName"));
            Assert.AreEqual(selenium.GetAttribute("name=name1@class"), "a2");
            // class location
            Assert.AreEqual(selenium.GetText("class=a3"), "this is the third element");
            // alt location
            Assert.IsTrue(selenium.IsElementPresent("alt=banner"));
            // identifier location
            Assert.AreEqual(selenium.GetText("identifier=id1"), "this is the first element");
            Assert.IsFalse(selenium.IsElementPresent("identifier=id4"));
            Assert.AreEqual(selenium.GetAttribute("identifier=id1@class"), "a1");
            Assert.AreEqual(selenium.GetText("identifier=name1"), "this is the second element");
            Assert.AreEqual(selenium.GetAttribute("identifier=name1@class"), "a2");
            // DOM Traversal location
            Assert.AreEqual(selenium.GetText("dom=document.links[1]"), "this is the second element");
            Assert.AreEqual(selenium.GetText("dom=function foo() {return document.links[1];}; foo();"),
                "this is the second element");
            Assert.AreEqual(selenium.GetText("dom=function foo() {\nreturn document.links[1];};\nfoo();"),
                "this is the second element");
            Assert.AreEqual(selenium.GetAttribute("dom=document.links[1]@class"), "a2");
            Assert.IsFalse(selenium.IsElementPresent("dom=document.links[9]"));
            Assert.IsFalse(selenium.IsElementPresent("dom=foo"));
            // Link location
            Assert.IsTrue(selenium.IsElementPresent("link=this is the second element"));
            Assert.IsTrue(selenium.IsTextPresent("this is the second element"));
            Assert.IsTrue(selenium.IsElementPresent("link=this * second element"));
            Assert.IsTrue(selenium.IsElementPresent("link=regexp:this [aeiou]s the second element"));
            Assert.AreEqual(selenium.GetAttribute("link=this is the second element@class"), "a2");
            Assert.IsFalse(selenium.IsElementPresent("link=this is not an element"));
            // SEL-484: IE: Can't select element by ID when there's another earlier element whose "name"
            // matches the ID
            Assert.IsTrue(selenium.IsElementPresent("name=foobar"));
            Assert.IsTrue(selenium.IsElementPresent("id=foobar"));
            // SEL-608:
            // "ID selector does not work when an element on the page has a name parameter equal to id"
            Assert.IsTrue(selenium.IsElementPresent("id=myForm"));
        }
    }
}

File: java/test/com/thoughtworks/selenium/corebased/TestLocators.java
// Licensed to the Software Freedom Conservancy (SFC) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The SFC licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

package com.thoughtworks.selenium.corebased;

import com.thoughtworks.selenium.InternalSelenseTestBase;

import org.junit.jupiter.api.Test;

public class TestLocators extends InternalSelenseTestBase {
  @Test
  public void testLocators() {
    selenium.open("test_locators.html");
    // Id location
    verifyEquals(selenium.getText("id=id1"), "this is the first element");
    verifyFalse(selenium.isElementPresent("id=name1"));
    verifyFalse(selenium.isElementPresent("id=id4"));
    verifyEquals(selenium.getAttribute("id=id1@class"), "a1");
    // name location
    verifyEquals(selenium.getText("name=name1"), "this is the second element");
    verifyFalse(selenium.isElementPresent("name=id1"));
    verifyFalse(selenium.isElementPresent("name=notAName"));
    verifyEquals(selenium.getAttribute("name=name1@class"), "a2");
    // class location
    verifyEquals(selenium.getText("class=a3"), "this is the third element");
    // alt location
    verifyTrue(selenium.isElementPresent("alt=banner"));
    // identifier location
    verifyEquals(selenium.getText("identifier=id1"), "this is the first element");
    verifyFalse(selenium.isElementPresent("identifier=id4"));
    verifyEquals(selenium.getAttribute("identifier=id1@class"), "a1");
    verifyEquals(selenium.getText("identifier=name1"), "this is the second element");
    verifyEquals(selenium.getAttribute("identifier=name1@class"), "a2");
    // DOM Traversal location
    verifyEquals(selenium.getText("dom=document.links[1]"), "this is the second element");
    verifyEquals(selenium.getText("dom=function foo() {return document.links[1];}; foo();"),
        "this is the second element");
    verifyEquals(selenium.getText("dom=function foo() {\nreturn document.links[1];};\nfoo();"),
        "this is the second element");
    verifyEquals(selenium.getAttribute("dom=document.links[1]@class"), "a2");
    verifyFalse(selenium.isElementPresent("dom=document.links[9]"));
    verifyFalse(selenium.isElementPresent("dom=foo"));
    // Link location
    verifyTrue(selenium.isElementPresent("link=this is the second element"));
    assertTrue(selenium.isTextPresent("this is the second element"));
    verifyTrue(selenium.isElementPresent("link=this * second element"));
    verifyTrue(selenium.isElementPresent("link=regexp:this [aeiou]s the second element"));
    verifyEquals(selenium.getAttribute("link=this is the second element@class"), "a2");
    verifyFalse(selenium.isElementPresent("link=this is not an element"));
    // SEL-484: IE: Can't select element by ID when there's another earlier element whose "name"
    // matches the ID
    verifyTrue(selenium.isElementPresent("name=foobar"));
    verifyTrue(selenium.isElementPresent("id=foobar"));
    // SEL-608:
    // "ID selector does not work when an element on the page has a name parameter equal to id"
    verifyTrue(selenium.isElementPresent("id=myForm"));
  }
}
